js/plugins/mcp/README.md

```diff
@@ -13,53 +15,94 @@ To get started, you'll need Genkit and the MCP plugin:
+      serverProcess: {
+        command: 'npx',
+        args: ['-y', '@modelcontextprotocol/server-everything', ...ALLOWED_DIRS],
+      },
+    },
+    git: { // Configuration for a second MCP server (hypothetical)
+      serverUrl: 'http://localhost:8080/mcp',
+      // disabled: true, // Optionally disable a server
+    },
   },
 });
 
 const ai = genkit({
   plugins: [
-    filesystemClient /* ... other plugins such as model providers ...*/,
+    /* ... install plugins such as model providers ...*/
   ],
 });
+
+const mcpTools = await mcpManager.getAllTools(ai);
+
+// Provide MCP tools to the model of your choice.
+const response = await ai.generate({
+  model: gemini15Flash,
+  prompt: 'What are the last 5 commits in the repo `/home/test-repo/?`',
+  tools,
+});
 ```
 
+The `createMcpClientManager` function initializes a `GenkitMcpClientManager` instance, which handles the lifecycle and communication with the defined MCP servers.
+
+### `createMcpClientManager()` Options
+
+-   **`name`**: (required, string) A name for the client manager plugin itself.
+-   **`version`**: (optional, string) The version of the client manager plugin. Defaults to "1.0.0".
+-   **`mcpClients`**: (required, object) An object where each key is a client-side name (namespace) for an MCP server, and the value is the configuration for that server.
+    Each server configuration object can include:
+    -   **`version`**: (optional, string) The specific version for this client.
+    -   **`rawToolResponses`**: (optional, boolean) If `true`, tool responses from this server are returned in their raw MCP format; otherwise, they are processed for Genkit compatibility. Defaults to `false`.
+    -   **`disabled`**: (optional, boolean) If `true`, this server connection will not be attempted. Defaults to `false`.
+    -   One of the following server connection configurations:
+        -   **`serverProcess`**: Parameters for launching a local server process using the stdio MCP transport.
+            -   **`command`**: (required, string) Shell command path for launching the MCP server (e.g., `npx`, `python`).
+            -   **`args`**: (optional, string[]) Array of string arguments to pass to the command.
+            -   **`env`**: (optional, Record<string, string>) Key-value object of environment variables.
+        -   **`serverUrl`**: (string) The URL of a remote server to connect to using the SSE MCP transport.
+        -   **`serverWebsocketUrl`**: (string) The URL of a remote server to connect to using the WebSocket MCP transport.
+        -   **`transport`**: An existing MCP transport object for connecting to the server.
+
 Most MCP servers are built to run as spawned processes on the same machine using the `stdio` transport. When you supply the `serverProcess` option, you are specifying the command, arguments, and environment variables for spawning the server as a subprocess.
 
-### mcpClient() Options
+### Legacy `mcpClient()`
```

Comment:
Let's mark `mcpClient` as deprecated but keep the old API, and have `createMcpClient` as not deprecated with the new API suggested above.